### PR TITLE
LF 4199: Fix undefined date range filter on Tasks page

### PR DESCRIPTION
**Pull request by Joaquin-M2**
**Description**

On the Tasks page, the date range filter threw `undefined` on the UI rather than applying the filter itself successfully.

Jira link: https://lite-farm.atlassian.net/browse/LF-4199

**Type of change**

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Passes test case
- [x] UI components visually reviewed on desktop view
- [x] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] The precommit and linting ran successfully  ---> The linting throws 11 errors and 14 warnings, but they are not related to changes made on this PR.
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

**Review by SayakaOno — COMMENTED** [review]
@Joaquin-M2 Thank you for working on this, your solution looks good!
It would be appreciated if you could look into the behaviour that an empty "To" filter is added:

https://github.com/LiteFarmOrg/LiteFarm/assets/33141219/54d6115f-0ddf-48c7-9b51-5fc23ffd74c9

File: packages/webapp/src/containers/Filter/Tasks/index.jsx [hunk 1]
```diff
@@ -166,6 +166,7 @@ const TasksFilterPage = ({ onGoBack }) => {
     },
     {
       subject: t('TASK.FILTER.DATE_RANGE'),
+      filterKey: FilterType.DATE_RANGE,
```
Comment (Duncan-Brain) [inline packages/webapp/src/containers/Filter/Tasks/index.jsx:169]: To keep it consistent with others should we get this from the constants file? The Filter types I could see being improved in the future.

File: packages/webapp/src/containers/Filter/types.ts [hunk 2]
```diff
@@ -20,6 +20,7 @@ export interface FilterState {
   [filterOptionId: string]: /* e.g. ABANDONED, COMPLETED */ {
     active?: boolean;
     label?: string /* e.g. 'Abandoned', 'Completed */;
+    [key: string]: any;
```
Comment (Duncan-Brain) [inline packages/webapp/src/containers/Filter/types.ts:23]: Ah it looks like this type is mostly used for multi-select right now. We might want to make the difference more explicit in the future!

**Review by Duncan-Brain — APPROVED** [review]
Works good for me. I left 2 small comments that I don't think need to be addressed right now. Awesome work!

**Review by SayakaOno — APPROVED** [review]
Thank you!

File: packages/webapp/src/containers/Filter/Tasks/index.jsx (hunk 1 above)
Comment (Joaquin-M2) [reply packages/webapp/src/containers/Filter/Tasks/index.jsx:169]: That's an option I considered too, but I wasn't sure if creating a new constant just for this was an overkill. I will create it if I work in a new ticket that requires that same value 👍

File: packages/webapp/src/containers/Filter/types.ts (hunk 2 above)
Comment (Joaquin-M2) [reply packages/webapp/src/containers/Filter/types.ts:23]: I can "defend" myself with TypeScript but I'm not any kind of TS wizard, so I start having troubles if typing things become a bit complex. I just wanted to get rid of the error 😆  But I agree that it should be more specific.

**Pull request merged**